ibm/service/database/resource_ibm_database.go

```diff
@@ -2800,12 +2874,19 @@ func validateGroupsDiff(_ context.Context, diff *schema.ResourceDiff, meta inter
 				}
 			}
 
-			if group.CPU != nil {
+			if (group.HostFlavor == nil || group.HostFlavor.ID == "" || group.HostFlavor.ID == "multitenant") && group.CPU != nil {
 				err = validateGroupScaling(groupId, "cpu", group.CPU.Allocation, groupDefaults.CPU, nodeCount)
 				if err != nil {
 					return err
 				}
 			}
+
+			if group.HostFlavor != nil && group.HostFlavor.ID != "" && group.HostFlavor.ID != "multitenant" {
```

Comment:
seems like you can validate this before other resources for the non-multi-tenant case, then checking for multi tenancy wouldn't be necessary for other resources

Reply:
Makes sense. Pushed an update